Joss paper review

samgdotson opened the pull request:
The paper lives in `paper/paper.md`. I would appreciate a review before submitting to JOSS!

Details about the submission process may be found at [joss.readthedocs.io](https://joss.readthedocs.io/en/latest/submitting.html).

Review criteria are available [here](https://joss.readthedocs.io/en/latest/review_criteria.html).

The content guidelines are:

- [ ] A list of the authors of the software and their affiliations, using the correct format (see the example below).
- [ ] A summary describing the high-level functionality and purpose of the software for a diverse, non-specialist audience.
- [ ] A Statement of need section that clearly illustrates the research purpose of the software and places it in the context of related work.
- [ ] A list of key references, including to other software addressing related needs. Note that the references should include full names of venues, e.g., journals and conferences, not abbreviations only understood in the context of a specific discipline.
- [ ] Mention (if applicable) a representative set of past or ongoing research projects using the software and recent scholarly publications enabled by it.
- [ ] Acknowledgement of any financial support.

samgdotson committed:
adds new install page

samgdotson committed:
Merge branch 'main' of github.com:samgdotson/osier into docs-examples

samgdotson committed:
adds instructions to install lp solvers

samgdotson committed:
adds more sphinx extensions

samgdotson committed:
adds nbsphinx and a first notebook tutorial

samgdotson committed:
adds pandoc

samgdotson committed:
adds line in the tech tutorial

samgdotson committed:
removes problematic build command

samgdotson committed:
updates navigation

samgdotson committed:
adds dispatch model tutorial

samgdotson committed:
updates the dispatch model tutorial

samgdotson committed:
adds capacity expansion notebook

samgdotson committed:
adds cap expansion tutorial to navigation

samgdotson committed:
first joss paper commit

samgdotson committed:
updates joss paper

samgdotson committed:
updates dispatch model tutorial

samgdotson committed:
Merge branch 'docs-examples' of github.com:samgdotson/osier into docs-examples

samgdotson committed:
add pandoc to setup.py

Co-authored-by: Olek <45364492+yardasol@users.noreply.github.com>

samgdotson committed:
changes initial problem to sizing natural gas and a battery

samgdotson committed:
adds bib file and updates state of the art

samgdotson committed:
updates dispatch tutorial

samgdotson committed:
big updates to osier paper

samgdotson committed:
adds sphinx-autobuild to doc reqs

samgdotson committed:
finishes tech_tutorial

samgdotson committed:
adds solver param to capacityexpansion

samgdotson committed:
updates tutorials

samgdotson committed:
Merge branch 'docs-examples' of github.com:samgdotson/osier into docs-examples

samgdotson committed:
updates capacity expansion tutorial

samgdotson committed:
completes the capacity expansion notebook

samgdotson committed:
removes pyomo from setup.py and adds guidance to readme

samgdotson committed:
tries to format note

samgdotson committed:
links issue

samgdotson committed:
installs pyomo in github workflow

samgdotson committed:
fixes line too long

samgdotson committed:
specifies correct pyomo version, awkward

samgdotson committed:
now is the line short enough?

samgdotson committed:
accidentally deleted pymoo instead of pyomo, awkward x2

samgdotson committed:
tells readthedocs to install pyomo

samgdotson committed:
tries another readthedocs cfg

samgdotson committed:
take 8 come on docs

samgdotson committed:
increases toc depth

samgdotson committed:
adds @smpark7 suggestions

Co-authored-by: Sun Myung Park <park_cg@live.com>

samgdotson committed:
updates dispatch and tech tutorials

samgdotson committed:
clarifies the ordering of the wind profile creation

samgdotson committed:
adds github workflow to compile paper

samgdotson committed:
updates bibfile?

samgdotson committed:
updates bib again

samgdotson committed:
fixes author metadata

samgdotson committed:
tries fixing citations

samgdotson committed:
moves paper docs to folder

samgdotson committed:
fixes paper path in CI?

samgdotson committed:
adds image to osier

samgdotson committed:
removes example code

samgdotson committed:
first full draft

samgdotson committed:
actual first draft

samgdotson committed:
fixes citations and line lengths

samgdotson committed:
minor edits

samgdotson committed:
addresses some review comments from @yardasol and @ZoeRichter

samgdotson committed:
Update CI.yml

samgdotson committed:
Merge branch 'main' of github.com:arfc/osier into docs-examples

samgdotson committed:
fixes note rendering

samgdotson committed:
removes extra note statement, it will render on readthedocs but not render on github, now.

samgdotson committed:
Merge pull request #5 from arfc/main

Update CI.yml

samgdotson committed:
Merge branch 'main' of github.com:samgdotson/osier into docs-examples

samgdotson committed:
Merge pull request #52 from samgdotson/main

Merge pull request #5 from arfc/main

samgdotson committed:
Merge branch 'main' of github.com:samgdotson/osier into docs-examples

samgdotson committed:
adds .gitattributes

samgdotson commented:
@munkm nudge?

samgdotson commented:
@munkm nudge

samgdotson commented:
@munkm nudge

samgdotson commented:
@munkm If the lack of PDF is an obstacle, here is the PDF. Thanks!
[paper.pdf](https://github.com/arfc/osier/files/14638518/paper.pdf)

smpark7 committed:
Merge pull request #45 from samgdotson/docs-examples

Docs examples

samgdotson commented:
@munkm nudge

munkm commented on .github/workflows/draft-pdf.yml:
```diff
@@ -0,0 +1,23 @@
+          journal: joss
+          # This should be the path to the paper within your repo.
+          paper-path: paper/paper.md
+      - name: Upload
+        uses: actions/upload-artifact@v1
+        with:
+          name: paper
+          # This is the output path where Pandoc will write the compiled
+          # PDF. Note, this should be the same directory as the input
+          # paper.md
```
I'm pretty sure you don't need the PDF rendering CI to submit. Do you really want this in the PR? Openjournals has a bot that will rerender it for you and then upload it into a issue/PR comment.

munkm replied:
This is not blocking though, just something to consider.

samgdotson replied:
It was definitely necessary when I was writing it up. Some changes would break the compilation.

munkm commented on paper/paper.md:
```diff
@@ -0,0 +1,127 @@
+recognition of the people and communities directly affected by this transition.
+However, public preferences may be ignored during decision-making processes
+related energy infrastructure due to a lack of technical rigor or expertise
+[@johnson:2021]. This challenge is further complicated by the fact that people
```
Is this the exact language from johnson? Would it be more appropriate to say this is a perceived lack of technical rigor or expertise? 

```suggestion
related to energy infrastructure due to a lack of technical rigor or expertise
[@johnson:2021]. This challenge is further complicated by the fact that people
```

munkm commented on paper/paper.md:
```diff
@@ -0,0 +1,127 @@
+Multi-objective optimization offers a method to help decision makers and
+stakeholders understand the problem and analyze tradeoffs among solutions
+[@liebman:1976]. Although, to date, no multi-objective energy modelling
+frameworks exist. Open-source multi-objective energy system framework (`osier`)
```
I thought the one in France existed? Would it be more appropriate to say no open source multi-objective energy modelling frameworks exist?

munkm commented on paper/paper.md:
```diff
@@ -0,0 +1,127 @@
+communities articulate their energy preferences in a technical manner without
+requiring extensive technical expertise. In order to facilitate more robust
+tradeoff analysis, `osier` generates a set of  technology portfolios, called a
+Pareto front, with multi-objective optimization using evolutionary algorithms.
```
The ordering of this phrasing sounds like the technology portfolios are called a pareto front, not that the pareto front defines the technology portfolios. What about something like this? 

```suggestion
tradeoff analysis, `osier` generates a set of solutions, called a
Pareto front, that are composed of a number of technology portfolios. The Pareto front is calculated using multi-objective optimization using evolutionary algorithms.
```

munkm commented on paper/paper.md:
```diff
@@ -0,0 +1,127 @@
+tradeoff analysis, `osier` generates a set of  technology portfolios, called a
+Pareto front, with multi-objective optimization using evolutionary algorithms.
+`osier` also extends the common
+modelling-to-generate-alternatives (MGA) algorithm into many dimensions,
```
```suggestion
modelling-to-generate-alternatives (MGA) algorithm into the dimensions of the solution space,
```

munkm commented on paper/paper.md:
```diff
@@ -0,0 +1,127 @@
+`osier` also extends the common
+modelling-to-generate-alternatives (MGA) algorithm into many dimensions,
+allowing users to investigate the near-optimal for appealing alternative
+solutions. In this way, `osier` may aid modelers in addressing procedural and
```
```suggestion
solutions. This is novel in that most energy system modeling tools apply MGA to a single dimension. In this way, `osier` may aid modelers in addressing procedural and
```

munkm commented on paper/paper.md:
```diff
@@ -0,0 +1,127 @@
+modelling-to-generate-alternatives (MGA) algorithm into many dimensions,
+allowing users to investigate the near-optimal for appealing alternative
+solutions. In this way, `osier` may aid modelers in addressing procedural and
+recognition justice.
```
Because the former is procedural, should this be recognitional?

munkm commented on paper/paper.md:
```diff
@@ -0,0 +1,127 @@
+There are myriad open- and closed-source energy system optimization models
+(ESOMs) available [@pfenninger:2022]. ESOMs can be used for a variety of tasks
+but are most frequently used for prescriptive analyses meant to guide
+decision-makers in planning processes. However, despite the many available
+models, all of these tools share a fundamental characteristic: Optimization over
```
```suggestion
decision-makers in planning processes. However, despite the countless 
models available, all of these tools share a fundamental characteristic: Optimization over
```

samgdotson replied:
I think "countless" is a little too hyperbolic.

munkm commented on paper/paper.md:
```diff
@@ -0,0 +1,127 @@
+Simultaneously, there is growing awareness of energy justice and calls for its
+inclusion in energy models [@pfenninger:2014; @vagero:2023]. Some studies
+attempted to incorporate local preferences into energy system design through
+multi-criteria decision analysis (MCDA) and community focus groups
```
What is the difference between MCDA and a multi objective energy modeling tool?

samgdotson replied:
MCDA focuses more on deciding among alternatives -- the way the alternatives are produced could be through MOO but they could also be generated through specific scenario analysis. At least, this is my understanding of the literature.

munkm commented on paper/paper.md:
```diff
@@ -0,0 +1,127 @@
+need has not yet been developed. The `osier` framework closes this gap.
+
+# Design and Implementation
+In order to run `osier`, users are only required to supply an energy demand time
```
```suggestion
In order to run `osier`, users are required to supply an energy demand time
```

munkm commented on paper/paper.md:
```diff
@@ -0,0 +1,127 @@
+
+# Design and Implementation
+In order to run `osier`, users are only required to supply an energy demand time
+series. Users can optionally provide weather data to incorporate solar or wind
```
Is this the only thing they are required to provide? What about what technologies they want to use and the limitations of the model?

munkm commented on paper/paper.md:
```diff
@@ -0,0 +1,127 @@
+
+# Acknowledgements
+
+Samuel Dotson was supported by the Nuclear Regulatory Commission Fellowship
```
Your acknowledgements should include the people who reviewed your PRs for Osier.

munkm replied:
Use the credit taxonomy to define who did what for this research output: https://credit.niso.org/
https://theplosblog.plos.org/2022/01/contributorship/

munkm commented on paper/paper.md:
```diff
@@ -0,0 +1,127 @@
+
+Samuel Dotson was supported by the Nuclear Regulatory Commission Fellowship
+program. This research was part of the Advanced Reactors and Fuel Cycles (ARFC)
+group in the Department of Nuclear, Plasma, and Radiological Engineering (NPRE)
+at the University of Illinois Urbana-Champaign. Additionally, Samuel Dotson was
+supported by the Felix T. Adler Fellowship through NPRE. The views and opinions
+expressed in this paper are those of the author alone.
```
```suggestion
group in the Department of Nuclear, Plasma, and Radiological Engineering (NPRE)
at the University of 
Illinois Urbana-Champaign. To that end, the authors would like to acknowledge ARFC 
members Oleksander Yardas, Luke Seifert, Nathan Ryan, Amanda Bachmann, 
and Sun Myung Park for their contributions in reviewing pull requests 
supporting the creation of osier. 
Additionally, Samuel Dotson was
supported by the Felix T. Adler Fellowship through NPRE. 
Finally, the authors would like to thank the JOSS reviewers for their time 
and commentary in reviewing this manuscript. 
```

munkm commented on paper/paper.md:
```diff
@@ -0,0 +1,127 @@
+
+# Acknowledgements
+
+Samuel Dotson was supported by the Nuclear Regulatory Commission Fellowship
```
```suggestion
Samuel Dotson, the corresponding and lead author of this publication is responsible 
for the conceptualization of osier, developing the osier as a software, in preparing 
this manuscript for publication, and for performing analysis to validate osier. 
Madicken Munk provided resources and supervision for the work, as well as assisted 
in the review and editing of the manuscript. Samuel Dotson was supported 
by the Nuclear Regulatory Commission Fellowship
```

munkm commented on paper/paper.md:
```diff
@@ -0,0 +1,127 @@
+      affiliation: 1 
+      corresponding: true
+affiliations:
+    - name: Felix T. Adler Fellow, Department of Nuclear, Plasma, and Radiological Engineering, University of Illinois Urbana-Champaign, USA
```
I don't believe your fellowship in an affiliation. It's part of your title. 

```suggestion
    - name: Department of Nuclear, Plasma, and Radiological Engineering, University of Illinois Urbana-Champaign, USA
```

munkm commented on paper/paper.md:
```diff
@@ -0,0 +1,127 @@
+    - genetic algorithms
+    - multi-objective optimization
+authors:
+    - name: Samuel G. Dotson
+      orcid: 0000-0002-8662-0336
+      affiliation: 1 
+      corresponding: true
+affiliations:
```
```suggestion
    - name: Samuel G. Dotson
      orcid: 0000-0002-8662-0336
      affiliation: 1 
      corresponding: true
    - name: Madicken Munk
      orcid: 0000-0003-0117-5366
      affiliation: 1 
affiliations:
```

munkm commented on paper/paper.md:
```diff
@@ -0,0 +1,127 @@
+affiliations:
+    - name: Felix T. Adler Fellow, Department of Nuclear, Plasma, and Radiological Engineering, University of Illinois Urbana-Champaign, USA
+      index: 1
+date: 20 February 2024
```
I think you should change this date to the date when you submit to JOSS.

munkm reviewed (changes requested):
I think this is in a good state. I've added inline suggestions, questions, and comments where necessary. One thing that I was curious about was that the focus currently of the paper is about what osier is and how it is built, but an aspect of the paper is that it (the paper and the software) should be understandable to a general audience. What do you think about adding a short section on explaining what the results of osier look like and how to interpret them? That is, an explanation about the pareto front solutions all being equally "good" but satisfying constraints differently? 

I also think your current PhD work is a great example of current research enabled by this code. I think you should add/mention that to support checkbox 5. 

- [x]  A list of the authors of the software and their affiliations, using the correct format (see the example below).
- [x]  A summary describing the high-level functionality and purpose of the software for a diverse, non-specialist audience.
- [x]  A Statement of need section that clearly illustrates the research purpose of the software and places it in the context of related work.
- [x]  A list of key references, including to other software addressing related needs. Note that the references should include full names of venues, e.g., journals and conferences, not abbreviations only understood in the context of a specific discipline.
- [ ]  Mention (if applicable) a representative set of past or ongoing research projects using the software and recent scholarly publications enabled by it.
- [x]  Acknowledgement of any financial support.

samgdotson committed:
Apply some suggestions from code review

Co-authored-by: Madicken Munk <madicken.munk@gmail.com>

samgdotson committed:
Merge branch 'main' of github.com:samgdotson/osier into joss

samgdotson committed:
Merge branch 'main' of github.com:arfc/osier into joss

samgdotson committed:
adds results section and @munkm suggestions

pep8speaks commented:
Hello @samgdotson! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:



* In the file [`osier/models/capacity_expansion.py`](https://github.com/arfc/osier/blob/fbcf05327ea9667cf12d70e831ca1f872cb00673/osier/models/capacity_expansion.py):

> [Line 53:80](https://github.com/arfc/osier/blob/fbcf05327ea9667cf12d70e831ca1f872cb00673/osier/models/capacity_expansion.py#L53): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)
> [Line 65:80](https://github.com/arfc/osier/blob/fbcf05327ea9667cf12d70e831ca1f872cb00673/osier/models/capacity_expansion.py#L65): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)





##### Comment last updated at 2024-04-04 22:12:21 UTC

samgdotson committed:
adds comment about results.

samgdotson commented:
@munkm I addressed your comments and believe this is ready to merge. If this gets merged into `main` I will remove the CI command to compile the paper -- but for now it's helpful to see that it builds correctly.

samgdotson commented:
@munkm here is the new pdf
[paper.pdf](https://github.com/arfc/osier/files/14871889/paper.pdf)

samgdotson commented:
Also, the PR ballooned because I pulled down the updates from `arfc:main`. This PR is being merged into `arfc:joss` which is not up-to-date with main. You can safely ignore any files that are not associated with the paper.

samgdotson committed:
Update README.md

samgdotson committed:
updates version and changelog

samgdotson committed:
Merge pull request #53 from samgdotson/main

updates version and changelog

samgdotson committed:
Merge branch 'main' of github.com:arfc/osier into joss

munkm commented on paper/paper.md:
```diff
@@ -0,0 +1,157 @@
+classes. `osier` comes pre-loaded with a variety of technologies described in 
+the National Renewable Energy Laboratory's (NREL) Annual Technology Baseline (ATB)
+dataset[@nationalrenewableenergylaboratory:2023] but users are also able to
+define their own. In order to run `osier`, users are only required to supply an 
```
For some reason this was missed in inclusion in the last round. `only` is not adding to your messaging here and makes your language less formal. 

```suggestion
define their own. In order to run `osier`, users are required to supply an 
```

munkm commented on paper/paper.md:
```diff
@@ -0,0 +1,157 @@
+own objectives and modify technology objects (or simply build their own by
+inheriting from the `osier.Technology` class) accounts for this limitation and
+expands the potential for incorporating localized preferences. Lastly, in order
+to account for unmodeled or unmodelable objectives, `osier` extends the
+conventional MGA algorithm into N-dimensions by using a farthest-first-traversal
+in the design space.
```
This is a good rephrase!

munkm commented on paper/paper.md:
```diff
@@ -0,0 +1,157 @@
+Transitioning to a clean energy economy will require expanded energy
+infrastructure. An equitable, or just, transition further requires the
+recognition of the people and communities directly affected by this transition.
+However, public preferences may be ignored during decision-making processes
+related to energy infrastructure due to a lack of technical rigor or expertise
+[@johnson:2021]. This challenge is further complicated by the fact that people
```
Is this the exact language from Johnson? Is it due to an actual lack of rigor or expertise, or a perceived lack of rigor or expertise? Would it be more appropriate to say a perceived lack of technical rigor or expertise?

I think this is important because this tool allows them to address the perception specifically, but doesn't make any assumptions about the expertise that they have or do not have.

samgdotson replied:
Here's a quote 

> While there may be other reasons the ICC never responded to Tripp’s allegations, her lack of expertise combined with the use of improper testimony helped ICC Staff sidestep the issue (Johson, 2021 p.7).

munkm replied:
Ok, then we can leave it. I do think for your future publications we should consider using language around osier and your tooling that clarifies its use in targeting a perception of lack of technical rigor.

munkm commented on paper/paper.md:
```diff
@@ -0,0 +1,157 @@
+\autoref{fig:osier-results} shows a Pareto front from a problem that
+simultaneously minimizes total cost and lifecylce carbon emissions.
+
+![A Pareto front generated by`osier`.\label{fig:osier-results}](images/osier-results.png)
+
+Each point on this Pareto front represents a different technology portfolio
+(i.e., different combination of wind, natural gas, and battery storage).
+\autoref{fig:osier-tech-res} illustrates the variation in solutions from 
+the Pareto front in \autoref{fig:osier-results}. In this case, the range of 
+wind capacity is wider than the range of capacities for natural gas and battery 
+storage.
+
+![The variance in technology options along a Pareto front.\label{fig:osier-tech-res}](images/osier-tech-results.png)
```
These added figures are good illustrations/additions. Very nice.

munkm reviewed (approved):
I have two minor questions, but the rest of this looks good. I like the additions you've made since the last round of revisions.

samgdotson committed:
Update paper/paper.md

Co-authored-by: Madicken Munk <madicken.munk@gmail.com>

munkm commented:
Thank you for the clarifications Sam. Looks great. Go ahead and submit!
